massgov/mayflower
# DP-25181 [a11y] Data visualization skip link target

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)


## Description

Ensure the skip data visualization link target gets focus when its corresponding link gets clicked.

## Related Issue / Ticket

- [JIRA issue](https://massgov.atlassian.net/browse/DP-25181)
- [Github issue]()

## Steps to Test


1. Go to /patterns/05-pages-information-details-dataviz/05-pages-information-details-dataviz.html
2. Display console to check the document.activeElement.
3. Tab through the page to navigate to one of the data visualization skip link.
<img width="1011" alt="1" src="https://user-images.githubusercontent.com/9633303/186182325-de1f08b7-34c0-4766-8f85-ae31d545b794.png">

5. Click the link, then find 

    - its corresponding landing target is displayed,
    - the target doesn't get covered by the sticky TOC,
    - the target is **document.activeElement**
    - the garget has visual focus indicator (blue outline)

<img width="1012" alt="2" src="https://user-images.githubusercontent.com/9633303/186182372-117ae6a2-6907-4991-8f64-ce2ce926b113.png">

6. Hit TAB or any arrow keys (attempt to leave the target), then find

    - the target loses focus,
    - the focus disappears,
    - the focus has now `tabindex="-1"`

<img width="1014" alt="3" src="https://user-images.githubusercontent.com/9633303/186182400-d168fc4d-e766-422f-9767-d42c41a3129d.png">

## Screenshots

Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


*

#### @TODO


*

#### Today I learned...
